spec/factories/samples.rb

```diff
@@ -26,16 +26,12 @@
 
     # metadata fields
     # ensure the metadata field is added to the host genome
-    before(:create) do |sample, options|
+    before(:create) do |_sample, options|
       options.metadata_fields.each_key do |metadata_field_name|
         metadata_field = MetadataField.find_by(name: metadata_field_name)
         unless metadata_field
-          metadata_field = create(:metadata_field, name: metadata_field_name)
+          create(:metadata_field, name: metadata_field_name)
```

Comment:
I'm glad we got rid of the `default_for_new_host_genome` hack! But don't we still need the `sample.host_genome.metadata_fields << metadata_field` line?